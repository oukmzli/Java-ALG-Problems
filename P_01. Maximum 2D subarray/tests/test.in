 // 1: Test na obsługę liczb dodatnich.
 // 2: Test na obsługę liczb ujemnych(test na "is empty").
 // 3: Test na obsługę elementów zerowych.
 // 4: Test na skomplikowaną kombinację liczb dodatnich i ujemnych.
 // 5: Test na dużą macierz z różnorodnymi liczbami.
 // 6: Test na jednorodne elementy.
 // 7: Podstawowy test na małej macierzy z naprzemiennym rozmieszczeniem.
 // 8: Test na średniej macierzy z równomiernym naprzemiennością.
 // 9: Test na dużej macierzy z ekstremalnymi wartościami.
 // 10: Złożony test na dużej macierzy z wyraźnym centrum.
 // 11: Test na dużej macierzy z pełnym naprzemiennością.
 11
 
 1: 2 2
 1 2
 3 4
 
 2: 3 3
 -1 -2 -3
 -4 -5 -6
 -7 -8 -9
 
 3: 2 3
 0 0 0
 0 0 0
 
 4: 3 4
 1 -1 2 -2
 3 -3 4 -4
 5 -5 6 -6
 
 5: 4 5
 1 2 -1 -4 5
 -1 3 4 -2 -3
 2 -3 -4 5 1
 -4 2 3 -1 -2
 
 6: 2 2
 5 5
 5 5
 
 7: 2 2
 1 -1
 -1 1
 
 8: 3 3
 1 -1 1
 -1 1 -1
 1 -1 1
 
 9: 5 5
 100000 -100000 100000 -100000 100000
 -100000 100000 -100000 100000 -100000
 100000 -100000 100000 -100000 100000
 -100000 100000 -100000 100000 -100000
 100000 -100000 100000 -100000 100000
 
 10: 10 10
 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
 -1 2 2 2 2 2 2 2 2 -1
 -1 2 -1 -1 -1 -1 -1 -1 2 -1
 -1 2 -1 100 100 -1 -1 -1 2 -1
 -1 2 -1 100 100 -1 -1 -1 2 -1
 -1 2 -1 -1 -1 -1 -1 -1 2 -1
 -1 2 2 2 2 2 2 2 2 -1
 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
 
 11: 10 10
 1 -1 1 -1 1 -1 1 -1 1 -1
 -1 1 -1 1 -1 1 -1 1 -1 1
 1 -1 1 -1 1 -1 1 -1 1 -1
 -1 1 -1 1 -1 1 -1 1 -1 1
 1 -1 1 -1 1 -1 1 -1 1 -1
 -1 1 -1 1 -1 1 -1 1 -1 1
 1 -1 1 -1 1 -1 1 -1 1 -1
 -1 1 -1 1 -1 1 -1 1 -1 1
 1 -1 1 -1 1 -1 1 -1 1 -1
 -1 1 -1 1 -1 1 -1 1 -1 1