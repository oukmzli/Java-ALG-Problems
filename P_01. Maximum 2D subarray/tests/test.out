1: ms_tab = a[0..1][0..1], msum=30
2: ms_tab is empty
3: ms_tab = a[0..0][0..0], msum=0
4: ms_tab = a[0..2][0..2], msum=45
5: ms_tab = a[0..3][1..4], msum=35
6: ms_tab = a[0..1][0..1], msum=60
7: ms_tab = a[0..0][0..0], msum=3
8: ms_tab = a[0..2][0..2], msum=7
9: ms_tab = a[0..4][0..4], msum=1500000
10: ms_tab = a[1..6][1..8], msum=1304
11: ms_tab = a[0..9][0..9], msum=50